python/src/overloaded.h

```diff
@@ -0,0 +1,8 @@
+// Copyright © 2023 Apple Inc.
+
+template <typename... Ts>
+struct overloaded : Ts... {
+  using Ts::operator()...;
+};
+template <class... Ts>
+overloaded(Ts...) -> overloaded<Ts...>;
```

Comment:
Does this need to be in a separate header? Can it be in linalg.cpp directly?

Reply:
Maybe we want to reuse this for 'pattern matching'. This is why I went for `overloaded.h`. If you think it not worth having it as a separate header, we can move to `linalg.cpp`.